nautobot/extras/datasources/git.py

```diff
@@ -908,7 +911,49 @@ def delete_git_config_context_schemas(repository_record, job_result, preserve=()
 
 def refresh_git_jobs(repository_record, job_result, delete=False):
     """Callback function for GitRepository updates - refresh all Job records managed by this repository."""
-    # No-op as jobs are not currently stored in the DB but are instead refreshed on-request.
+    installed_jobs = []
+    if not delete:
+        jobs_path = os.path.join(repository_record.filesystem_path, "jobs")
+        if os.path.isdir(jobs_path):
+            for job_info in jobs_in_directory(jobs_path):
+                job_model, created = refresh_job_model_from_job_class(
+                    Job,
+                    f"{JobSourceChoices.SOURCE_GIT}.{repository_record.slug}",
+                    job_info.job_class,
+                )
+                if created:
+                    job_result.log(
+                        f'Created Job model "{job_model}"',
+                        grouping="jobs",
+                        level_choice=LogLevelChoices.LOG_SUCCESS,
+                        logger=logger,
+                    )
+                else:
+                    job_result.log(
+                        f'Refreshed Job model "{job_model}"',
+                        grouping="jobs",
+                        level_choice=LogLevelChoices.LOG_SUCCESS,
+                        logger=logger,
+                    )
+                installed_jobs.append(job_model)
+        else:
+            job_result.log(
+                "No `jobs` subdirectory found in Git repository",
+                grouping="jobs",
+                level_choice=LogLevelChoices.LOG_WARNING,
+                logger=logger,
+            )
+
+    for job_model in Job.objects.filter(source=f"{JobSourceChoices.SOURCE_GIT}.{repository_record.slug}"):
+        if job_model.installed and job_model not in installed_jobs:
```

Comment:
I foresee some huge documentation necessary with #275 if it lands. Having jobs work as expected but it gets removed in an automated sync...

Reply:
I may be missing what you're thinking here - any issue that would be present in an automated sync would also potentially occur during a manual sync as well. Can you elaborate on your concern?

Reply:
So with what's coming into `next` manual can be dry ran so you will know what will be changed. All automated will have some surprises. But I just wonder if the impression with a concrete model is that things might stay more because they are in the DB now. Just curious.

Reply:
Rephrasing to possibly make my thought clearer:

Pre-concrete model, I believe it's well understood that the list of jobs is a "no-op" because they "are not currently stored in the DB but are instead refreshed on-request".

Moving to a concrete model might be perceived by some that the job definition would stay around upon syncs, but do overridden settings get lost if the job disappears in the Git repo? What happens if a job is renamed in the repo? Manual syncs would make those changes but now that we will have a `dry-run` feature, we can expose "You are about to delete a job, a job that has overridden settings, is that desired?"

I don't think there is anything to change with regard to this PR to address this but adding the work of #275 sometime down the road might cause behaviors that should be called out, even if they are accepted.

Reply:
> Moving to a concrete model might be perceived by some that the job definition would stay around upon syncs, 

It does.

> but do overridden settings get lost if the job disappears in the Git repo?

Not as presently implemented - the JobModel database record just gets updated to `installed = False`.

> What happens if a job is renamed in the repo?

That would be detected as a new job and reflected as a new database record (the old database record under the old name would still exist but, again, be marked as `installed = False`) - we don't really have any way to detect renames as opposed to delete/create.

> Manual syncs would make those changes but now that we will have a `dry-run` feature, we can expose "You are about to delete a job, a job that has overridden settings, is that desired?"

Until and unless we get to the point of automatically deleting JobModel database records instead of just marking them as not-installed, this shouldn't be an issue.


To be clear these are all great questions - I probably need to update the Git docs to address these!